sky/skylet/providers/azure/node_provider.py

```diff
@@ -303,6 +303,37 @@ def _create_node(self, node_config, tags, count):
         template_params["nsg"] = self.provider_config["nsg"]
         template_params["subnet"] = self.provider_config["subnet"]
 
+        # pylint: disable=import-outside-toplevel
+        from sky.clouds.service_catalog import azure_catalog
+
+        instance_type = node_config["azure_arm_parameters"].get("vmSize", "")
+        accs = azure_catalog.get_accelerators_from_instance_type(instance_type)
+        if accs is not None and "A10" in accs:
```

Comment:
Is there a way to avoid using `azure_catalog` in the node_provider as that seems to be an abstraction leakage, but I am ok to keep it as is for now.

Reply:
Actually, how about in the `azure-ray.yaml` we pass a `need_nvidia_driver_extension: true` within `clouds/azure.py::make_deployable_variables`. There we can check the accelerators, and get rid of the use of catalog here.

Reply:
Done! PTAL 🫡